components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/APIPublisherImpl.java

```diff
@@ -620,9 +620,19 @@ public void updateCheckListItem(String apiId, String status, Map<String, Boolean
     public String createNewAPIVersion(String apiId, String newVersion) throws APIManagementException {
         String newVersionedId;
         LifecycleState lifecycleState;
+        if (StringUtils.isEmpty(newVersion)) {
+            String errorMsg = "New API version cannot be empty";
+            log.error(errorMsg);
+            throw new APIManagementException(errorMsg, ExceptionCodes.PARAMETER_NOT_PROVIDED);
+        }
         try {
             API api = getApiDAO().getAPI(apiId);
             if (api != null) {
+                if (api.getVersion().equals(newVersion)) {
+                    String errMsg = "New API version cannot be same as the previous version";
```

Comment:
Shall we add context information to the log. 

Example : "New API version" + newVersion  +  "cannot be same as the previous version" for API + APIName